### ISSUE TITLE
Stuck after successfully login via SAML

### ISSUE THREAD

**Issue by shidenko97** [yuezk/GlobalProtect-openconnect#46]
I have problem with app. It stops on "Connecting" state after successfully login.
My OS: Ubuntu 20.04
APP version: 1.2.7
Logs:
```
2021-03-05 07:44:12.016 INFO  [4607] [main@22] GlobalProtect started, version: v1.2.7
2021-03-05 07:44:12.128 INFO  [4607] [GPClient::populateGatewayMenu@100] Populating the Switch Gateway menu...
2021-03-05 07:44:22.326 INFO  [4607] [GPClient::doConnect@205] Start connecting...
2021-03-05 07:44:22.327 INFO  [4607] [GPClient::doConnect@226] Start portal login...
2021-03-05 07:44:22.333 INFO  [4607] [PortalAuthenticator::authenticate@29] Preform portal prelogin at https://***/global-protect/prelogin.esp?tmp=tmp&kerberos-support=yes&ipv6-support=yes&clientVer=4100&clientos=Linux
2021-03-05 07:44:22.337 INFO  [4607] [GPClient::populateGatewayMenu@100] Populating the Switch Gateway menu...
2021-03-05 07:44:22.881 INFO  [4607] [PortalAuthenticator::onPreloginFinished@46] Portal prelogin succeeded.
2021-03-05 07:44:22.881 INFO  [4607] [PreloginResponse::parse@26] Start parsing the prelogin response...
2021-03-05 07:44:22.882 INFO  [4607] [PortalAuthenticator::onPreloginFinished@50] Finished parsing the prelogin response. The region field is: UA
2021-03-05 07:44:22.882 INFO  [4607] [PortalAuthenticator::samlAuth@117] Trying to perform SAML login with saml-method POST
2021-03-05 07:44:22.978 INFO  [4607] [SAMLLoginWindow::onResponseReceived@64] Response received from ***
2021-03-05 07:44:22.992 INFO  [4607] [SAMLLoginWindow::onLoadFinished@98] Load finished https://***/global-protect/prelogin.esp?tmp=tmp&kerberos-support=yes&ipv6-support=yes&clientVer=4100&clientos=Linux
2021-03-05 07:44:23.008 INFO  [4607] [GPClient::populateGatewayMenu@100] Populating the Switch Gateway menu...
2021-03-05 07:44:23.539 INFO  [4607] [SAMLLoginWindow::onResponseReceived@64] Response received from https://***/app/panw_globalprotect/***/sso/saml
2021-03-05 07:44:23.687 INFO  [4607] [SAMLLoginWindow::onLoadFinished@98] Load finished https://***/app/panw_globalprotect/***/sso/saml
2021-03-05 07:44:23.699 INFO  [4607] [SAMLLoginWindow::onResponseReceived@64] Response received from https://***/discovery/iframe.html
2021-03-05 07:44:27.904 INFO  [4607] [SAMLLoginWindow::onResponseReceived@64] Response received from https://***/auth/services/devicefingerprint
2021-03-05 07:44:29.441 INFO  [4607] [SAMLLoginWindow::onResponseReceived@64] Response received from https://***/login/sessionCookieRedirect
2021-03-05 07:44:29.503 INFO  [4607] [SAMLLoginWindow::onLoadFinished@98] Load finished https://***/login/sessionCookieRedirect
2021-03-05 07:44:29.508 INFO  [4607] [SAMLLoginWindow::onResponseReceived@64] Response received from https://***/discovery/iframe.html
2021-03-05 07:44:38.455 INFO  [4607] [SAMLLoginWindow::onResponseReceived@64] Response received from https://***/login/sessionCookieRedirect
2021-03-05 07:44:38.499 INFO  [4607] [SAMLLoginWindow::onLoadFinished@98] Load finished https://***/login/sessionCookieRedirect
2021-03-05 07:44:38.889 INFO  [4607] [SAMLLoginWindow::onResponseReceived@64] Response received from https://***/SAML20/SP/ACS
2021-03-05 07:44:38.890 INFO  [4607] [SAMLLoginWindow::onResponseReceived@67] Got username from SAML response headers ***
2021-03-05 07:44:38.890 INFO  [4607] [SAMLLoginWindow::onResponseReceived@72] Got prelogin-cookie from SAML response headers ***
2021-03-05 07:44:38.890 INFO  [4607] [SAMLLoginWindow::onResponseReceived@84] Got the SAML authentication information successfully. username: ***, preloginCookie: ***, userAuthCookie: 
2021-03-05 07:44:38.890 INFO  [4607] [PortalAuthenticator::onSAMLLoginSuccess@131] SAML login succeeded, got the prelogin-cookie ***
2021-03-05 07:44:38.890 INFO  [4607] [PortalAuthenticator::fetchConfig@157] Fetching the portal config from https://***/global-protect/getconfig.esp for user: ***
2021-03-05 07:44:38.897 INFO  [4607] [GPClient::populateGatewayMenu@100] Populating the Switch Gateway menu...
2021-03-05 07:44:39.373 INFO  [4607] [PortalAuthenticator::onFetchConfigFinished@183] Fetch the portal config succeeded.
2021-03-05 07:44:39.373 INFO  [4607] [PortalConfigResponse::parse@20] Start parsing the portal configuration...
2021-03-05 07:44:39.374 INFO  [4607] [PortalConfigResponse::parseGateways@64] Start parsing the gateways from portal configuration...
2021-03-05 07:44:39.374 INFO  [4607] [PortalConfigResponse::parsePriorityRules@96] Start parsing the priority rules...
2021-03-05 07:44:39.374 INFO  [4607] [PortalConfigResponse::parsePriorityRules@114] Finished parsing the priority rules.
2021-03-05 07:44:39.374 INFO  [4607] [PortalConfigResponse::parseGatewayName@121] Start parsing the gateway name...
2021-03-05 07:44:39.374 INFO  [4607] [PortalConfigResponse::parseGatewayName@126] Finished parsing the gateway name
2021-03-05 07:44:39.374 INFO  [4607] [PortalConfigResponse::parseGateways@89] Finished parsing the gateways.
2021-03-05 07:44:39.374 INFO  [4607] [PortalConfigResponse::parse@32] Start reading portal-userauthcookie
2021-03-05 07:44:39.374 INFO  [4607] [PortalConfigResponse::parse@35] Start reading portal-prelogonuserauthcookie
2021-03-05 07:44:39.375 INFO  [4607] [PortalConfigResponse::parse@42] Finished parsing portal configuration.
2021-03-05 07:44:39.375 INFO  [4607] [GPClient::onPortalSuccess@257] Portal authentication succeeded.
2021-03-05 07:44:39.375 INFO  [4607] [gpclient::helper::filterPreferredGateway@34] 1 gateway(s) avaiable, filter the gateways with rule: UA
2021-03-05 07:44:39.375 INFO  [4607] [GPClient::setAllGateways@384] Updating all the gateways...
2021-03-05 07:44:39.375 INFO  [4607] [GPClient::populateGatewayMenu@100] Populating the Switch Gateway menu...
2021-03-05 07:44:39.377 INFO  [4607] [GPClient::setCurrentGateway@404] Updating the current gateway to ***
2021-03-05 07:44:39.377 INFO  [4607] [GPClient::populateGatewayMenu@100] Populating the Switch Gateway menu...
2021-03-05 07:44:39.378 INFO  [4607] [GPClient::gatewayLogin@316] Performing gateway login...
2021-03-05 07:44:39.380 INFO  [4607] [GatewayAuthenticator::authenticate@26] Start gateway authentication...
2021-03-05 07:44:39.380 INFO  [4607] [GatewayAuthenticator::login@38] Trying to login the gateway at https://***/ssl-vpn/login.esp with prot=https%3A&server=&inputSrc=&jnlpReady=jnlpReady&computer=***&ok=Login&direct=yes&clientVer=4100&os-version=Ubuntu 20.04.2 LTS&clientos=Linux&portal-prelogonuserauthcookie=&prelogin-cookie=&ipv6-support=yes&user=***&passwd=&portal-userauthcookie=empty
2021-03-05 07:44:39.875 ERROR [4607] [GatewayAuthenticator::onLoginFinished@49] Failed to login the gateway at https://***/ssl-vpn/login.esp, Error transferring https://***/ssl-vpn/login.esp - server replied: Custom error
2021-03-05 07:44:39.876 INFO  [4607] [GatewayAuthenticator::doAuth@70] Perform the gateway prelogin at https://***/ssl-vpn/prelogin.esp?tmp=tmp&kerberos-support=yes&ipv6-support=yes&clientVer=4100&clientos=Linux
2021-03-05 07:44:40.090 INFO  [4607] [GatewayAuthenticator::onPreloginFinished@87] Gateway prelogin succeeded.
2021-03-05 07:44:40.090 INFO  [4607] [PreloginResponse::parse@26] Start parsing the prelogin response...
2021-03-05 07:44:40.091 INFO  [4607] [GatewayAuthenticator::samlAuth@145] Trying to perform SAML login with saml-method POST
2021-03-05 07:44:40.114 INFO  [4607] [SAMLLoginWindow::onResponseReceived@64] Response received from ***
2021-03-05 07:44:40.132 INFO  [4607] [SAMLLoginWindow::onLoadFinished@98] Load finished https://***/ssl-vpn/prelogin.esp?tmp=tmp&kerberos-support=yes&ipv6-support=yes&clientVer=4100&clientos=Linux
2021-03-05 07:44:40.158 INFO  [4607] [GPClient::populateGatewayMenu@100] Populating the Switch Gateway menu...
2021-03-05 07:44:40.503 INFO  [4607] [SAMLLoginWindow::onResponseReceived@64] Response received from https://***/app/panw_globalprotect/***/sso/saml
2021-03-05 07:44:40.557 INFO  [4607] [SAMLLoginWindow::onLoadFinished@98] Load finished https://***/app/panw_globalprotect/***/sso/saml
2021-03-05 07:44:40.673 INFO  [4607] [SAMLLoginWindow::onResponseReceived@64] Response received from https://***/SAML20/SP/ACS
2021-03-05 07:44:40.673 INFO  [4607] [SAMLLoginWindow::onResponseReceived@67] Got username from SAML response headers ***
2021-03-05 07:44:40.673 INFO  [4607] [SAMLLoginWindow::onResponseReceived@72] Got prelogin-cookie from SAML response headers ***
2021-03-05 07:44:40.673 INFO  [4607] [SAMLLoginWindow::onResponseReceived@84] Got the SAML authentication information successfully. username: ***, preloginCookie: ***, userAuthCookie: 
2021-03-05 07:44:40.673 INFO  [4607] [GatewayAuthenticator::onSAMLLoginSuccess@159] SAML login succeeded, got the prelogin-cookie ***
2021-03-05 07:44:40.674 INFO  [4607] [GatewayAuthenticator::login@38] Trying to login the gateway at https://***/ssl-vpn/login.esp with prot=https%3A&server=&inputSrc=&jnlpReady=jnlpReady&passwd=&computer=***&ok=Login&direct=yes&clientVer=4100&os-version=Ubuntu 20.04.2 LTS&clientos=Linux&portal-prelogonuserauthcookie=&ipv6-support=yes&user=***&prelogin-cookie=***&portal-userauthcookie=
2021-03-05 07:44:40.681 INFO  [4607] [GPClient::populateGatewayMenu@100] Populating the Switch Gateway menu...
2021-03-05 07:44:41.110 INFO  [4607] [gpclient::helper::parseGatewayResponse@50] Start parsing the gateway response...
2021-03-05 07:44:41.110 INFO  [4607] [gpclient::helper::parseGatewayResponse@51] The gateway response is: ***
2021-03-05 07:44:41.110 INFO  [4607] [GPClient::onGatewaySuccess@330] Gateway login succeeded, got the cookie authcookie=***&portal=FullTunnel-N&user=***&domain=%2528empty_domain%2529&preferred-ip=&computer=***
```

**Comment by shidenko97** [comment]
I've found the problem - I hadn't open connect package(
But maybe it will be fine to display some error about that

**Comment by yuezk** [comment]
Fixed in #81 , now it will display an error dialog when OpenConnect not found.
